src/neuroglancer/util/http_request.ts

```diff
@@ -120,13 +120,77 @@ export function getByteRangeHeader(startOffset: Uint64|number, endOffset: Uint64
   return {'Range': `bytes=${startOffset}-${endOffsetStr}`};
 }
 
-export function parseUrl(url: string): {protocol: string, host: string, path: string} {
-  const urlProtocolPattern = /^([^:\/]+):\/\/([^\/]+)((?:\/.*)?)$/;
-  let match = url.match(urlProtocolPattern);
-  if (match === null) {
-    throw new Error(`Invalid URL: ${JSON.stringify(url)}`);
+export class ParsedUrl{
```

Comment:
Can we use the existing URL class already provided by browsers, or is it unsuitable for some reason?

https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
I figured eventually `ParsedUrl` (or whatever other name you might want to give it) could be used to handle URLs in more places throughout the code where URL manipulation is required. The native implementation would be insufficient because it silently breaks on funny protocols:

```
new URL("gs://example.com/some/path")
// produces URL { (...) host: "", hostname: "", port: "", pathname: "//example.com/some/path"}
// note the empty hostname, which was interpreted as being part of the pathname
```

So even if we extended `URL`, we'd still need to do the parsing ourselves.

Still, if we split handling the special Neuroglancer URLs (e.g. `precomputed://http://...`, `gs://`) from the expanded, canonical URLs (i.e.: `https?:`), then extending the native `URL` would be enough. But I think unifying all URL handling under a single class would be simpler to follow.